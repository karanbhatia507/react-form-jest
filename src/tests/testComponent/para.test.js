import React from 'react';
import {  mount } from 'enzyme';
import Paragraph from '../../components/para'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

configure({ adapter: new Adapter() })

describe('Testing Paragraph Component', () => {
    it('Paragraph component should be defined', () => {
        const component = mount(<Paragraph text="Hey paragraph" className="para" />)
        expect(component.prop('text')).toEqual("Hey paragraph")
        expect(component.prop('className')).toEqual("para")
    })
    it('Typeof of Props', () => {
        const component = mount(<Paragraph text="Hey paragraph" className="para" />)
        expect(typeof component.prop('text')).toBe("string")
        expect(component.prop('className')).toContain('para')
    })
    it('Snapshot Component tree', () => {
        const component = mount(<Paragraph text="Hey paragraph!" className="para" />)
        const tree = toJson(component)
        expect(tree).toMatchSnapshot()
    })
})