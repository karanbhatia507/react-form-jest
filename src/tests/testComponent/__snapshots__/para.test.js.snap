// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Paragraph Component Snapshot Component tree 1`] = `
<Paragraph
  className="para"
  text="Hey paragraph!"
>
  <p>
    Hey paragraph!
  </p>
</Paragraph>
`;
